src/Watchers/HttpClientWatcher.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+
+namespace Spatie\LaravelRay\Watchers;
+
+
+use Illuminate\Http\Client\Events\RequestSending;
+use Illuminate\Http\Client\Events\ResponseReceived;
+use Illuminate\Http\Client\Request;
+use Illuminate\Http\Client\Response;
+use Illuminate\Support\Facades\Event;
+use Spatie\Ray\Payloads\TablePayload;
+use Spatie\LaravelRay\Ray;
+use Spatie\Ray\Settings\Settings;
+use SplObjectStorage;
+
+class HttpClientWatcher extends Watcher
+{
+    protected SplObjectStorage $requestTimings;
+
+    public function __construct()
+    {
+        $this->requestTimings = new SplObjectStorage();
```

Comment:
Why not use a regular array?

Reply:
I'm storing the entire request object as the key, so that we can pull it out in the second listener. I did think about using the current microtime as the key instead, but that felt a little "iffy".

Reply:
I guess we could use an array of arrays, with the request as the first parameter and the microtime as the second, and then loop through to find the instance, but at that point, are you really saving anything?